### PR TITLE
Feat/random median blur

### LINKED ISSUE #941: [Feat] Kornia Augmentations tracker
## More Than Augmentation
In a longer vision, our augmentation module will be more focused on differentiability, with an aim of assisting both training and deployment. We can see there are more researches and applications in this area, with or without Kornia. This module will be the lead. In our next version, we gonna heavily improve:
- the API design and better readability.
- Advanced features like JIT and AMP.
- Better differentiability and more differentiable augmentations.

## 🚀 Feature
Specs for the `kornia.augmentation` module.

- Make it easy to contrib through `AugmentationBase2D`.
- Give support to `torch.jit.script` and `torch.float16`.
- Promote to embed the full code in the main class to keep readability.
- Documentation improvement:
  - Better looking
  - More examples for copying and pasting.
  - Provide a benchmark to notify users that we are better.
- Add `register_buffer` to save static parameters inside ```nn.Module```.

## Motivation

This should be the template to implement a custom augmentation

```python
class RandomHorizontalFlip(AugmentationBase2D):

    def compute_transformation(self, input: torch.Tensor) -> torch.Tensor:
        w: int = input.shape[-1]
        flip_mat: torch.Tensor = torch.tensor([[-1, 0, w - 1],
                                               [0, 1, 0],
                                               [0, 0, 1]], device=input.device, dtype=input.dtype)
        return flip_mat.repeat(input.size(0), 1, 1)

    def apply_transform(self, input: torch.Tensor, params: Dict[str, torch.Tensor]) -> torch.Tensor:
        return kornia.hflip(input)
```

## User Stories
- _As a user, I want …, so that …_ 
- As a user, I want to decide whether the transforms return the transformation, so that I can use it to chain and potentially apply in sequential.
- As a user, I want to use the augmentations with half precision, so that I can also combine with AMP.
- As a user, I want to be able to run the augmentations with xla devices, so that I can run on TPUs.
- As a user, I want to be able to torchscript the augmentations, so that I can easily serialise and export.
- As a developer, I want to have easy examples of custom augmentations, so that I can contribute with more.

## TODO:
- [x] Refactor engine to deprecate `return_transform` to return only `torch.Tensor` @shijianjian 
  - Transforms will be accessible through class properties
- [ ] Enable torchscript for augmentations https://github.com/kornia/kornia/issues/477
- [x] Remove functional code @shijianjian 
- [x] Fix `RandResizedCrop` critical bug https://github.com/kornia/kornia/issues/927

## List of existing 2d transforms:
- [x] CenterCrop
- [x] ColorJitter
- [x] GaussianBlur
- [x] RandomAffine
- [x] RandomCrop
- [ ] RandomErasing
  - Gradients error. They computed on the non-cropped area than on the mask. Check the gradient image in the bottom. will be fixed in https://github.com/kornia/kornia/pull/873 
- [x] RandomGrayscale
- [x] RandomHorizontalFlip
- [x] RandomVerticalFlip
- [x] RandomMotionBlur
- [x] RandomPerspective
- [x] RandomResizedCrop
  - Broken: https://github.com/kornia/kornia/issues/927
- [x] RandomRotation
- [x] RandomSolarize
- [ ] RandomPosterize
  - Not differentiable (due to the basis of int8 operations). Grad_fn not implemented.
- [x] RandomSharpness
- [ ] RandomEqualize
  - Not backward-propagatable
  - Error message: “derivative for floor_divide is not implemented”
- [x] RandomMixUp
- [x] RandomCutMix

## List of wanted 2d transforms:
- **Adapt from Tormentor:** needs tormentor’s diamond square
  - [x] RandomelasticTransform  @edgarriba
  - [x] RandomThinPlateSpline  @edgarriba
  - [x] RandomFishEye  @edgarriba
    - Adopt from tormentor: https://github.com/anguelos/tormentor#create-custom-augmentation-class
  - [x] RandomPlasmaBrightness
  - [x] RandomPlasmaContrast
  - [ ] RandomPlasmaRgbBrightness
  - [x] RandomPlasmaShadow
  - [ ] ImageCorruption augmentations https://github.com/kornia/kornia/issues/1143
- **Adapt from Albumentations**
  - [x] RandomInvert https://github.com/kornia/kornia/pull/960 @edgarriba 
  - [x] RandomGaussianNoise https://github.com/kornia/kornia/pull/960 @edgarriba 
  - [x] RandomChannelShuffle https://github.com/kornia/kornia/pull/960 @edgarriba 
  - [ ] RandomGlassNoise
  - [x] RandomRGBShift
  - [ ] RandomClahe
  - [x] RandomBoxBlur
  - [ ] RandomMedianBlur
  - [ ] RandomChannelDropout
  - [ ] RandomDownScale
  - [ ] RandomRain -> need `draw_line`
  - [ ] RandomShadow -> need `draw_line`
  - [ ] RandomSnow https://github.com/UjjwalSaxena/Automold--Road-Augmentation-Library/blob/master/Automold.py#L330
  - [ ] RandomSunFlare -> need `draw_line`
- **Containers**
  - [x] VideoSequential
  - [x] OneOf ? or RandomChoice
  - [x] ToTensor https://github.com/kornia/kornia/issues/705
  - [x] Lambda
  - [x] RandomApply
  - [ ] AutoAugment ? (or similar) @shijianjian 
  - [ ] TTA @shijianjian 
  - [ ] Augmentation presets https://github.com/kornia/kornia/issues/1151

Link to discussion docs: https://docs.google.com/document/d/1DcWnoB5L-Ws0asOq9ntXHrCe0ny2NM2WCtLIoU50wQw/edit?usp=sharing

## Advanced features
 - [x] `AugmentationSequential` to support multi-data augmentations
 - [ ] Make `AugmentationSequential` end to end. 
     - [x] Bounding Box #1497 
     - [ ] Mask
     - [x] Keypoints
     - [ ] Label
- [x] Deprecate `return_transform` in favour of `inverse`.
- [ ] AutoAugment pipelines.

-----

## Please, anyone interested: comment below and volunteer yourself :smile:

### PR DESCRIPTION
**Pull request by DariaMinieieva**
#### Changes

Add RandomMedianBlur to the augmentation module

Fixes #941 - RandomMedianBlur

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [x] 📝 This change requires a documentation update

#### Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: kornia/augmentation/_2d/intensity/median_blur.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+from torch import Tensor
+
+from kornia.augmentation._2d.intensity.base import IntensityAugmentationBase2D
+from kornia.filters import median_blur
+
+
+class RandomMedianBlur(IntensityAugmentationBase2D):
+    """Add random blur with a median filter to an image tensor.
+
+    .. image:: _static/img/RandomMedianBlur.png
```
Comment (edgarriba) [inline kornia/augmentation/_2d/intensity/median_blur.py:12]: You need to update this file
https://github.com/kornia/kornia/blob/master/docs/generate_example_images.py

**Review by shijianjian — APPROVED** [review]
